src/Bicep.Core/Workspaces/SourceFileFactory.cs

```diff
@@ -25,14 +25,19 @@ public static class SourceFileFactory
 
         public static ISourceFile CreateSourceFile(Uri fileUri, string fileContents, ModuleReference? moduleReference = null)
         {
+            if (PathHelper.HasBicepExtension(fileUri))
+            {
+                return CreateBicepFile(fileUri, fileContents);
+            }
+
             if (PathHelper.HasArmTemplateLikeExtension(fileUri))
             {
                 return moduleReference is TemplateSpecModuleReference
                     ? CreateTemplateSpecFile(fileUri, fileContents)
                     : CreateArmTemplateFile(fileUri, fileContents);
             }
 
-            return CreateBicepFile(fileUri, fileContents);
+            return CreateAuxiliaryTextFile(fileUri, fileContents);
```

Comment:
I think we need to handle the case where the user explicitly sets the language mode to `Bicep` in VS Code for an unsaved new file (or any file that doesn't have the Bicep extension). Otherwise, the language server will do nothing:
![bicep-viz-live-update2](https://user-images.githubusercontent.com/16367959/164556505-b41da989-7c37-43e2-8b2e-cb68d6359321.gif)

Reply:
I looked into how other language servers handle this case, and, while it's possible to recognized unsaved buffers by their URI scheme (`untitled:` for VS Code, `unsaved:` for Nova, and probably a few more for the other editors listed on [the LSP tools page](https://microsoft.github.io/language-server-protocol/implementors/tools/)), it would help avoid unwanted side effects (like the one you found) to keep this check in the build command handler. Similar checks are performed in the CLI or the language server rather than in Bicep.Core, so I think this fits better with our overall validation strategy.

One possible drawback to the minimal approach is that module paths are permitted to use any extension, so this PR would not completely resolve #5091.

Reply:
I looked into how other language servers handle this case, and, while it's possible to recognized unsaved buffers by their URI scheme (`untitled:` for VS Code, `unsaved:` for Nova, and probably a few more for the other editors listed on [the LSP tools page](https://microsoft.github.io/language-server-protocol/implementors/tools/)), it would help avoid unwanted side effects (like the one you found) to keep this check in the build command handler. Similar checks are performed in the CLI or the language server rather than in Bicep.Core, so I think this fits better with our overall validation strategy.

One possible drawback to the minimal approach is that module paths are permitted to use any extension, so this PR would not completely resolve #5091.

Reply:
In the language server we can probably just check the `languageId` property in the `textDocument/didOpen` requests to determine if the document is a Bicep file:

```
[Trace - 10:45:23 AM] Sending notification 'textDocument/didOpen'.
Params: {
    "textDocument": {
        "uri": "untitled:Untitled-1",
        "languageId": "bicep",
        "version": 1,
        "text": ""
    }

```

Reply:
For the language server, that check ends up happening in the BicepCompilationManager to prevent the server from trying to compile arbitrary text files: 

https://github.com/Azure/bicep/blob/888968e7c33cbffe1e3973a69ff4147770fc58f1/src/Bicep.LangServer/BicepCompilationManager.cs#L180-L187

With a languageId check in the language server and a file extension check in the build command, I believe the only way a non-bicep file ends up getting passed to the core compiler is if it's the source of a module (which is a pretty clear user signal that the file should be treated as Bicep source (or ARM if it has the right extension)).